types/react/index.d.ts

```diff
@@ -897,7 +897,27 @@ declare namespace React {
         onTransitionEndCapture?: TransitionEventHandler<T>;
     }
 
-    export interface CSSProperties extends CSS.Properties<string | number> {}
+    export interface CSSProperties extends CSS.Properties<string | number> {
+        /**
+         * The index signature was removed to enable closed typing for style
+         * using CSSType. You're able to use module augmentation to add other
+         * properties, like CSS Custom Properties, or add a index signature
+         * of your own.
+         *
+         * ```ts
+         * // my-react.d.ts
+         * import * as React from 'react';
+         *
+         * declare module 'react' {
+         *   interface CSSProperties {
+         *     '--my-custom-property': string;
+         *     // or
+         *     [index: string]: any;
+         *   }
+         * }
+         * ```
```

Comment:
Maybe also good to mention that a cheap alternative, esp. for missing properties where you don't need to reference it in multiple places, is casting:

```ts
<div style={{ ['overflowAnchor' as any]: 'auto' }}>
  ...
</div>
```

Reply:
I added that to the README a  `csstype`. Thanks 👍